File: practicas/practica_1/Deepspace_Ruby_P1/lib/ShieldBooster.rb
#encoding: utf-8

# File: ShieldBooster.rb
# Author: Juan Manuel Rodríguez Gómez

module Deepspace

    # Brief: Representa los potenciadores de escudo que pueden tener las 
    #        estaciones espaciales.
    class ShieldBooster

        # Constructor con parámetros
        def initialize(p_name, p_boost, p_uses)
            @name = p_name
            @boost = p_boost
            @uses = p_uses
        end

        # Constructor de copia
        def self.newCopy(sb)
            copy = ShieldBooster.new(sb.name, sb.boost, sb.uses)
        end

        public
        
        # Consultor del nombre
        def name
            @name
        end

        # Consultor del valor del potenciador
        def boost
            @boost
        end
      
        # Consultor de los usos
        def uses
            @uses
        end

        # Emula el uso del potenciador de escudo.
        def useIt
            if(uses > 0)
              @uses -= 1
              return @boost
            else
              return 1.0
            end
        end
    end
end

File: practicas/practica_1/deepspace_Java_P1/src/main/java/deepspace/ShieldBooster.java
/**
 * @file ShieldBooster.java
 * @author Juan Manuel Rodríguez Gómez
 */
package deepspace;

/**
 * @brief Representa los potenciadores de escudo que pueden tener las 
 *        estaciones espaciales.
 */
class ShieldBooster {
    
    private String name;
    private float boost;
    private int uses;
    
    /**
     * @brief Constructor con parámetros
     * @param p_name Valor del nombre
     * @param p_boost Valor del potenciador
     * @param p_uses Valor de los usos
     */
    ShieldBooster(String p_name, float p_boost, int p_uses) {
        name = p_name;
        boost = p_boost;
        uses = p_uses;
    }
    
    /**
     * @brief Constructor de Copia
     * @param sb Otro objeto de la clase ShieldBooster
     */
    ShieldBooster(ShieldBooster sb) {
        name = sb.name;
        boost = sb.boost;
        uses = sb.uses;
    }
    
    /**
     * @brief Consultor del valor del potenciador
     * @return boost
     */
    public final float getBoost() {
        return boost;
    }
    
    /**
     * @brief Consultor de los usos
     * @return uses
     */
    public final int getUses() {
        return uses;
    }
    
    /**
     * @brief Emula el uso del potenciador de escudo.
     */
    public float useIt() {
        
        if(getUses() > 0) {
            uses--;
            return boost;
        }
        else
            return 1.0f;
    }
}